docs/example_workflow_explained.rst

```diff
@@ -0,0 +1,316 @@
+
+.. code:: ipython3     
+    
+    site_to_seg = { site_0_name : site_0_seg, ...} # Input this mapping or read it from a text file before running!
+
+Since it is nice to be able to access the data you just filled out without much struggle, here we create
+some other useful forms of these gauge site mappings for later use.
+
+.. code:: ipython3 
+
+    seg_to_site = {seg: site for site, seg in site_to_seg.items()}
+    ref_sites = list(site_to_seg.keys())
+    ref_segs = list(site_to_seg.values())
+    
+Next we load in topographical data (topo), meterological data (met), 
+uncorrected flows (raw), and reference flows (ref). Note that some
+fields have placeholder names that you should update before running.
+If some data is not accessible in a single function call, be sure to collapse
+it into a single file first before loading them. File designation calls assume
+this code is in a folder seperate from the data, but that this code's containing
+folder is at the same heirarchy as the folders containing the data. A description
+of how your project directory is expected to be set up can be foudn in ``data.rst``.
+    
+.. code:: ipython3 
+    
+    basin_topo = xr.open_dataset('../topologies/basin_topology_file_name.nc').load() 
+    
+Sometimes meterological data may only be available for a larger region
+or watershed than anlayzing, so the following data will be described under such
+an assumption.
+    
+Here we load in some example meterological data: minimum temperature (tmin), seasonal precipitation (prec),
+and maximum temperature (tmax). You can use similar or completely different data, just note naming should be universally updated and unused names should be deleted or commented out completely.
+
+.. code:: ipython3 
+
+    watershed_met = xr.open_dataset('../input/tmin.nc').load()
+    watershed_met['seasonal_precip'] = xr.open_dataset('../input/prec.nc')['prec'].load().rolling(time=30, min_periods=1).sum()
+    watershed_met['tmax'] = xr.open_dataset('../input/tmax.nc')['tmax'].load()
+    
+Hydrualic residence units (hru's) are the typical coordinate for meteorlogical data. Later, mizuroute_utils
+will take care of mapping these hru's to seg's.
+    
+.. code:: ipython3 
+    
+    watershed_met['hru'] = (watershed_met['hru'] - 1.7e7).astype(np.int32)
+    
+And last not be certainly not least, we need the flows themselves! ``bmorph`` is a post-processing method,
+meaning a streamflow routing through mizuroute should occur before running all of this. As a result, loading
+up the raw flows involves combining a number of flow netcdf files, hence the ``open_mfdataset``.
+
+.. code:: ipython3 
+
+    watershed_raw = xr.open_mfdataset('../input/first_route*.nc')[['IRFroutedRunoff', 'dlayRunoff', 'reachID']].load()
+    watershed_raw['seg'] = watershed_raw.isel(time=0)['reachID'].astype(np.int32)
+    watershed_ref = xr.open_dataset('../input/nrni_reference_flows.nc').load().rename({'outlet':'site'})[['seg', 'seg_id', 'reference_flow']]
```

Comment:
I would split some of these into multiple commands for new users to be able to see each step separately a bit better. Not a requirement for the PR just a note that users might not be super familiar with xarray.

Reply:
raised as issue, since I bet there are some other areas this could be improved too